### PR TITLE
fix: merge route custom protocol

### PR DESCRIPTION
**Pull request by Donghui0**
### Description
When Route configures service_protocols and service_id attributes at the same time, merge_service_route does not take effect

fix plugin.merge_service_route function.

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
Hi, the test cases is needed

File: t/node/grpc-proxy.t [hunk 1]
```diff
@@ -285,3 +285,36 @@ grpcurl -import-path ./t/grpc_server_example/proto -proto helloworld.proto -plai
 qr/grpc header: "(:authority|host): [^"]+"/
 --- grep_error_log_out eval
 qr/grpc header: "(:authority|host): hello.world"/
+
+
+=== TEST 10: with service_id
```
Comment (monkeyDluffy6017) [inline t/node/grpc-proxy.t:290]: The name of the test case needs to make sense

**Comment by monkeyDluffy6017** [comment]
Please make the ci pass

**Comment by soulbird** [comment]
we have already remove service_protocol： https://github.com/apache/apisix/pull/7701

**Comment by Donghui0** [comment]
> we have already remove service_protocol： #7701

why ?   There are scenarios where the same service provides multiple protocols.

**Comment by soulbird** [comment]
> > we have already remove service_protocol： #7701
> 
> why ? There are scenarios where the same service provides multiple protocols.

you can use upstream. scheme

**Comment by Donghui0** [comment]
> > > we have already remove service_protocol： #7701
> > 
> > 
> > why ? There are scenarios where the same service provides multiple protocols.
> 
> you can use upstream. scheme

same service  provides multiple protocols !!!

**Pull request closed without merging**